vms/platformvm/camino_service.go

```diff
@@ -762,61 +799,52 @@ func (s *CaminoService) GetDeposits(_ *http.Request, args *GetDepositsArgs, repl
 		if err != nil {
 			return err
 		}
-		reply.Deposits[i].UnlockableAmount = utilsjson.Uint64(deposit.UnlockableAmount(offer, timestamp))
-		reply.AvailableRewards[i] = utilsjson.Uint64(deposit.ClaimableReward(offer, timestamp))
+		reply.Deposits[i].UnlockableAmount = json.Uint64(deposit.UnlockableAmount(offer, timestamp))
+		reply.AvailableRewards[i] = json.Uint64(deposit.ClaimableReward(offer, timestamp))
 	}
 	return nil
 }
 
 // GetLastAcceptedBlock returns the last accepted block
-func (s *CaminoService) GetLastAcceptedBlock(r *http.Request, _ *struct{}, reply *api.GetBlockResponse) error {
-	s.vm.ctx.Log.Debug("Platform: GetLastAcceptedBlock called")
+func (s *CaminoService) GetLastAcceptedBlock(r *http.Request, args *api.JSONEncoding, reply *api.GetBlockResponse) error {
+	s.vm.ctx.Log.Debug("API called",
+		zap.String("service", "platform"),
+		zap.String("method", "getLastAcceptedBlock"),
+		zap.Stringer("encoding", args.Encoding),
+	)
+
+	s.vm.ctx.Lock.Lock()
+	defer s.vm.ctx.Lock.Unlock()
 
 	ctx := r.Context()
-	lastAcceptedID, err := s.vm.LastAccepted(ctx)
+	blockID, err := s.vm.LastAccepted(ctx)
 	if err != nil {
 		return fmt.Errorf("couldn't get last accepted block ID: %w", err)
 	}
-	block, err := s.vm.manager.GetStatelessBlock(lastAcceptedID)
+	block, err := s.vm.manager.GetStatelessBlock(blockID)
 	if err != nil {
-		return fmt.Errorf("couldn't get block with id %s: %w", lastAcceptedID, err)
+		s.vm.ctx.Log.Error("couldn't get accepted block",
+			zap.Stringer("blkID", blockID),
+			zap.Error(err),
+		)
+		return fmt.Errorf("couldn't get block with id %s: %w", blockID, err)
 	}
 
-	block.InitCtx(s.vm.ctx)
-	reply.Encoding = formatting.JSON
-	reply.Block = block
-	return nil
-}
-
-type GetBlockAtHeight struct {
-	Height uint32 `json:"height"`
-}
-
-// GetBlockAtHeight returns block at given height
-func (s *CaminoService) GetBlockAtHeight(r *http.Request, args *GetBlockAtHeight, reply *api.GetBlockResponse) error {
```

Comment:
Removed, because avax added their own impl, which uses latest features and way more performant.

Reply:
indeed, good call